dev/test/bulk-writer.ts

```diff
@@ -505,6 +506,63 @@ describe('BulkWriter', () => {
     return bulkWriter.close();
   });
 
+  describe.only('500/50/5 support', () => {
+    afterEach(() => setTimeoutHandler(setTimeout));
+
+    it('does not send batches if doing so exceeds the rate limit', done => {
+      // The test is considered a success if BulkWriter tries to send the second
+      // batch again after a timeout.
+      setTimeoutHandler((_, timeout) => {
+        // Check that BulkWriter has not yet sent the 2nd batch.
+        expect(requestCounter).to.equal(0);
+
+        expect(timeout).to.be.greaterThan(0);
+        done();
+      });
+
+      const arrayRange = Array.from(new Array(500), (_, i) => i);
+      const requests1 = arrayRange.map(i => setOp('doc' + i, 'bar'));
+      const responses1 = arrayRange.map(i => successResponse(i));
+      const arrayRange2 = [500, 501, 502, 503, 504];
+      const requests2 = arrayRange2.map(i => setOp('doc' + i, 'bar'));
+      const responses2 = arrayRange2.map(i => successResponse(i));
+
+      instantiateInstance(
+        [
+          {
+            request: createRequest(requests1),
+            response: mergeResponses(responses1),
+          },
+          {
+            request: createRequest(requests2),
+            response: mergeResponses(responses2),
+          },
+        ],
+        /* enforceSingleConcurrentRequest= */ true
```

Comment:
IMHO, you don't need to set this here, at which point you can also drop the call to `activeRequestDeferred`.

Reply:
thanks for the rec -- it really simplified the test.